File: 6_kyu/Sums_of_Parts.go
package kata

// Calculate the accumulated sums.
// Return an array with
// result[i] = sum(ls[i],ls[i+1],...,ls[last])
func PartsSums(ls []uint64) []uint64 {
	var sums []uint64 = make([]uint64, len(ls)+1)
	var accu uint64 = 0
	sums[len(ls)] = 0
	for i := len(ls) - 1; i >= 0; i-- {
		accu += ls[i]
		sums[i] = accu
	}
	return sums
}

File: 6_kyu/Sums_of_Parts.lua
local solution = {}

-- Calculates list of partial sums.
function solution.partsSums(ls)
  local sum = 0
  local sums = {}
  sums[#ls+1] = 0
  for i = #ls,1,-1 do
    sum = sum + ls[i]
    sums[i] = sum
  end
  return sums
end

return solution
